File: src/main/java/persistence/sql/ddl/table/Table.java
package persistence.sql.ddl.table;

import persistence.sql.ddl.column.Columns;

public class Table {

    private final TableName name;
    private final Columns columns;

    private Table(TableName name, Columns columns) {
        this.name = name;
        this.columns = columns;
    }

    public static Table from(Class<?> entity) {
        TableName name = TableName.from(entity);
        Columns columns = Columns.from(entity);

        return new Table(name, columns);
    }

    public String getName() {
        return name.getName();
    }

    public String getColumnsDefinition() {
        return columns.getColumnsDefinition();
    }

    public String getColumnsClause() {
        return columns.getColumnsClause();
    }
}

File: src/main/java/persistence/sql/ddl/table/Table.kt
package persistence.sql.ddl.table

import parse.ParseType
import parse.ParseType.NAME
import persistence.sql.find

class Table (
    private val tableParsers: Map<ParseType, TableParser>,
    private val clazz: Class<*>
) {


    fun createQuery(): String {
        return "CREATE TABLE ${tableParsers.find(NAME).parse(clazz)}"
    }

    fun dropQuery(): String {
        return "DROP TABLE ${tableParsers.find(NAME).parse(clazz)}"
    }
}
